Compiler error with warp 0.3.6

Under linux I get the following compiler error. Strangely enough it compiles fine under Windows. This seems to happen with Rust 1.71 and 1.72.

```
error[E0698]: type inside `async fn` body must be known in this context
   --> /cargo/registry/src/index.crates.io-6f17d22bba15001f/foxglove-ws-0.2.1/src/lib.rs:406:21
    |
406 |                     Ok(warp::reply::with_header(
    |                     ^^ cannot infer type for type parameter `E` declared on the enum `Result`
    |
note: the type is part of the `async fn` body because of this `await`
   --> /cargo/registry/src/index.crates.io-6f17d22bba15001f/foxglove-ws-0.2.1/src/lib.rs:413:44
    |
413 |         warp::serve(foxglove_ws).run(addr).await;
    |                                            ^^^^^

For more information about this error, try `rustc --explain E0698`.
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -397,17 +397,17 @@ impl FoxgloveWebSocket {
         let channels = warp::any().map(move || channels.clone());
         let foxglove_ws = warp::path::end().and(
             warp::ws()
-                .and(warp::filters::header::value("sec-websocket-protocol"))
                 .and(clients)
                 .and(channels)
-                .map(|ws: warp::ws::Ws, proto, clients, channels| {
-                    let reply =
-                        ws.on_upgrade(move |socket| client_connected(socket, clients, channels));
-                    Ok(warp::reply::with_header(
+                .map(|ws: warp::ws::Ws, clients, channels| {
+                    ws.on_upgrade(move |socket| client_connected(socket, clients, channels))
+                })
+                .map(|reply| {
+                    warp::reply::with_header(
                         reply,
-                        "sec-websocket-protocol",
-                        proto,
-                    ))
+                        "Sec-WebSocket-Protocol",
+                        "foxglove.websocket.v1",
+                    )
                 }),
         );
         warp::serve(foxglove_ws).run(addr).await;

